ic10emu/src/interpreter.rs

```diff
@@ -2590,3 +2600,83 @@ pub fn i64_to_f64(i: i64) -> f64 {
     }
     i as f64
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::vm::VMError;
+
+    use super::*;
+
+    #[test]
+    fn batch_modes() -> Result<(), VMError> {
+        let mut vm = VM::new();
+        let ic = vm.add_ic(None).unwrap();
+        let ic_id = {
+            let device = vm.devices.get(&ic).unwrap();
+            let device_ref = device.borrow();
+            device_ref.ic.unwrap()
+        };
+        let ic_chip = vm.ics.get(&ic_id).unwrap().borrow();
+        vm.set_code(
+            ic,
+            r#"lb r0 HASH("ItemActiveVent") On Sum
+            #lb r1 HASH("ItemActiveVent") On Maximum
+            lb r2 HASH("ItemActiveVent") On Minimum"#,
```

Comment:
should maybe use a prefab that actually is a prefab, I just picked whatever the autocomplete gave me when I made the issue 😅

Reply:
true, but then again the test is intentionally designed to look for the result where no devices were found